File: tests/benchmark/corpus/java/safe/SafeDirectPathSanitizer.java
// java-safe-014: direct-return path sanitiser using `.contains` / `.startsWith`.
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class SafeDirectPathSanitizer {
    public static String sanitizePath(String s) {
        if (s.contains("..") || s.startsWith("/") || s.startsWith("\\")) {
            return "";
        }
        return s;
    }

    public static void handle(String userPath) throws IOException {
        String safe = sanitizePath(userPath);
        Files.readAllBytes(Paths.get(safe));
    }
}

File: tests/benchmark/corpus/php/safe/safe_direct_path_sanitizer.php
<?php
// php-safe-014: direct-return path sanitiser using `strpos` / substring checks.

function sanitize_path(string $s): string {
    if (strpos($s, '..') !== false || $s[0] === '/' || $s[0] === '\\') {
        return '';
    }
    return $s;
}

function handle(string $userPath): void {
    $safe = sanitize_path($userPath);
    file_get_contents($safe);
}
?>
